goldens/public-api/forms/forms.d.ts

```diff
@@ -205,37 +207,39 @@ export declare class FormArrayName extends ControlContainer implements OnInit, O
 }
 
 export declare class FormBuilder {
-    array(controlsConfig: any[], validatorOrOpts?: ValidatorFn | ValidatorFn[] | AbstractControlOptions | null, asyncValidator?: AsyncValidatorFn | AsyncValidatorFn[] | null): FormArray;
-    control(formState: any, validatorOrOpts?: ValidatorFn | ValidatorFn[] | AbstractControlOptions | null, asyncValidator?: AsyncValidatorFn | AsyncValidatorFn[] | null): FormControl;
-    group(controlsConfig: {
-        [key: string]: any;
+    array<Item = any>(controlsConfig: FormControlConfig<Item>[], validatorOrOpts?: ValidatorFn | ValidatorFn[] | AbstractControlOptions | null, asyncValidator?: AsyncValidatorFn | AsyncValidatorFn[] | null): FormArray;
+    control<T = any>(formState: FormControlState<T>, validatorOrOpts?: ValidatorFn | ValidatorFn[] | AbstractControlOptions | null, asyncValidator?: AsyncValidatorFn | AsyncValidatorFn[] | null): FormControl;
+    group<T extends object = any>(controlsConfig: {
+        [key in keyof T]: FormControlConfig<T[key]>;
     }, options?: AbstractControlOptions | {
         [key: string]: any;
     } | null): FormGroup;
 }
 
-export declare class FormControl extends AbstractControl {
-    constructor(formState?: any, validatorOrOpts?: ValidatorFn | ValidatorFn[] | AbstractControlOptions | null, asyncValidator?: AsyncValidatorFn | AsyncValidatorFn[] | null);
-    patchValue(value: any, options?: {
+export declare class FormControl<T = any> extends AbstractControl<T> {
+    constructor(formState?: FormControlState<T>, validatorOrOpts?: ValidatorFn | ValidatorFn[] | AbstractControlOptions | null, asyncValidator?: AsyncValidatorFn | AsyncValidatorFn[] | null);
+    patchValue(value: null | T, options?: {
```

Comment:
Is it correct? As described in the docs, we might provide some fields we would like to update, not the full model structure. With these changes, we will have some TypeScript types errors.
```suggestion
    patchValue(value: null | Partial<T>, options?: {
```

Reply:
@destus90 , I think it's correct, `Partial<T>` need only for `FormGroup`. Please give an example with `FormControl` where `Partial<T>` really needs to be used.

Reply:
Ok, you are right. At first I thought these changes will affect to all the reactive directives.